capellambse/_diagram_cache.py

```diff
@@ -27,301 +28,196 @@
-            E.h1(
-                title,
-                E.span(" created: ", now, {"class": "small"}),
-            )
-        )
-        diaglist = E.ol()
-        for diagram in self._index:
-            label = diagram["name"]
-            uuid = E.span(diagram["uuid"], {"class": "uuid"})
-            if diagram["success"]:
-                href = f"./{diagram['uuid']}.{self.image_format}"
-                diaglist.append(E.li(E.a(label, href=href), " ", uuid))
-            else:
-                diaglist.append(E.li(label, uuid, {"class": "missing"}))
-        html.append(diaglist)
-        filepath.write_text(ET.tostring(html).decode("ascii"), encoding="utf8")
-
-    def _post_process_svg(self, path: pathlib.Path) -> None:
-        """Post process SVG file.
-
-        Parameters
-        ----------
-        path
-            Absolute path to the SVG file of an exported diagram
-        """
-        tree = ET.parse(path)
-
-        for elm in tree.iter():
-            attrib = elm.attrib
-            if "font-family" in attrib:
-                attrib[
-                    "font-family"
-                ] = "'Open Sans','Segoe UI',Arial,sans-serif"
-            if "stroke-miterlimit" in attrib:
-                del attrib["stroke-miterlimit"]
-
-        path.write_bytes(ET.tostring(tree, pretty_print=True))
-
-    def export_diagrams(self, project_dir: pathlib.Path) -> None:
-        with tempfile.TemporaryDirectory() as workspace:
-            build_cli_cmd = getattr(self, f"_build_cli_cmd_{self._cli_type}")
-            build_cli_cmd(project_dir, workspace)
-            try:
-                subprocess.check_call(self._cli_cmd)
-            except subprocess.CalledProcessError as err:
-                raise RuntimeError("Failed to update diagram cache") from err
-            for p in pathlib.Path(project_dir).glob(
-                f"main_model/**/*.{self.image_format}"
-            ):
-                shutil.copyfile(p, self.diagrams_dir / p.name)
-        self._create_index_json()
-        self._rename_diagram_image_files()
-        if self.image_format == "svg":
-            for svg in pathlib.Path(self.diagrams_dir).glob("*.svg"):
-                self._post_process_svg(svg)
-        if self.create_index:
-            self._json_index_path.write_text(json.dumps(self._index, indent=2))
-            self._create_index_html_file()
+        ),
+        E.body(
+            E.h1(title, E.span({"class": "small"}, " created: ", now)),
+            ol := E.ol(),
+        ),
+    )
+
+    for diagram in sorted(index, key=operator.itemgetter("name")):
+        uuid = E.span(diagram["uuid"], {"class": "uuid"})
+        if diagram["success"]:
+            href = f"{diagram['uuid']}.{extension}"
+            label = E.a({"href": href}, diagram["name"])
+        else:
+            label = E.span({"class": "missing"}, diagram["name"])
+        ol.append(E.li(label, " ", uuid))
+
+    dest.joinpath("index.json").write_text(json.dumps(index))
```

Comment:
```suggestion
    dest.joinpath("index.json").write_text(json.dumps(index, indent=2))
```
Let us write a human-readable file here. A oneliner is useless when one needs to read it. With syntax highlighting enabled it takes quite a while to format the file with a LSP. That can be annoying.

Reply:
When we talk about 22kB or even 20 times that size I do not see the point to discuss network traffic here.
When we want to check a produced JSON file (a scenario I expect to happen) it makes sense, that we can just read it w/o processing.
I totally agree (minifying stuff) when the size requires it. Here, the files will be very small (even with > 1,000 diagrams).
I formatted with the attached LSP `vscode-json-language-server` and the command lua vim.lsp.buf.format().
takes long for an index.json that includes diagram element information having a size of 136 kB and 5,400 lines (when formatted)